diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -58,6 +59,101 @@ int DifferentialFunctionComparator::cmpGEPs(const GEPOperator *GEPL,
         // The original function says the GEPs are equal - return the value
         return OriginalResult;
 
+    if (isa<ArrayType>(GEPL->getSourceElementType())
+        && isa<ArrayType>(GEPR->getSourceElementType())) {
+        // The number of elements of two enums can be different and
+        // its not considered as semantic change
+        if (GEPL->getSourceElementType()->getArrayNumElements()
+            == GEPR->getSourceElementType()->getArrayNumElements())
+            return OriginalResult;
+
+        if (int Res =
+                    cmpNumbers(GEPL->getNumOperands(), GEPR->getNumOperands()))
+            return Res;
+
+        for (unsigned i = 0, e = GEPL->getNumOperands(); i != e; ++i) {
+            if (int Res = cmpValues(GEPL->getOperand(i), GEPR->getOperand(i)))
+                return Res;
+        }
+        if (GEPL->hasAllConstantIndices() || GEPR->hasAllConstantIndices())
+            return OriginalResult;
+
+        auto instrL = dyn_cast<Instruction>(GEPL->getOperand(2));
+        auto instrR = dyn_cast<Instruction>(GEPR->getOperand(2));
```

Comment:
You should check that the GEPs have at least 3 instructions. Also, let's limit this case for GEPs that have exactly 3 instructions.

Reply:
Not sure, but it's done.